pinot-common/src/main/java/org/apache/pinot/sql/parsers/rewriter/ExprMinMaxRewriter.java

```diff
@@ -107,7 +108,8 @@ private void appendParentExprMinMaxFunctions(boolean isMax, List<Expression> sel
       operands.add(RequestUtils.getLiteralExpression((int) exprMinMaxFunctionIDMap.get(measuringColumns)));
       operands.add(RequestUtils.getLiteralExpression(measuringColumns.size()));
       operands.addAll(measuringColumns);
-      operands.addAll(projectionColumns);
+      Set<Expression> sortedProjectionColumns = new TreeSet<>(projectionColumns);
```

Comment:
This is adding overhead to production code. Since the set is in general small, you may directly create `TreeSet` at first place (on line 169). To be more concise, you may change the code as following:
```
    boolean added = exprMinMaxFunctionMap.computeIfAbsent(exprMinMaxMeasuringExpressions, k -> new TreeSet<>())
        .add(exprMinMaxProjectionExpression);
```